### PR TITLE
Column encryption

### PR DESCRIPTION
**Pull request by bgfoshay**
## What’s changing?

Created a shared library for encrypting and decrypting values with Azure Key Vault using the RSA algorithm. This library can be used for column level encryption.

## Why?

NAC-84

## This PR has:

- [x] **Commented source code** (_required on public classes and methods, and elsewhere as appropriate_)

- [ ] **Developer documentation** (_for new build/test/API changes or complex portions of the system_)

- [x] **Automated unit tests** (_to maintain or increase level of code coverage_)

- [ ] **Changes to IAC** (_add any deployment steps that require manual intervention to the draft release notes_)

### CODE REVIEW

File: shared/tests/Piipan.Shared.Cryptography.Tests/packages.lock.json [hunk 1]
```diff
@@ -0,0 +1,1305 @@
...
+        }
+      },
+      "xunit.extensibility.execution": {
+        "type": "Transitive",
+        "resolved": "2.4.0",
+        "contentHash": "252Dzn7i5bMPKtAL15aOP3qJhxKd+57I8ldwIQRJa745JxQuiBu5Da0vtIISVTtc3buRSkBwVnD9iUzsEmCzZA==",
+        "dependencies": {
+          "xunit.extensibility.core": "[2.4.0]"
+        }
+      }
```
Comment (dborgen2) [inline shared/tests/Piipan.Shared.Cryptography.Tests/packages.lock.json:1302]: You may want to see if you can regenerate this packages.lock file. When I built the project it added this to the bottom:
```
   "piipan.shared.cryptography": {
        "type": "Project",
        "dependencies": {
          "Azure.Identity": "1.5.0",
          "Azure.Security.KeyVault.Keys": "4.3.0"
        }
      }

File: shared/tests/Piipan.Shared.Cryptography.Tests/Piipan.Shared.Cryptography.Tests.csproj [hunk 2]
```diff
@@ -0,0 +1,24 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
```
Comment (dborgen2) [inline shared/tests/Piipan.Shared.Cryptography.Tests/Piipan.Shared.Cryptography.Tests.csproj:4]: Do we need the following lines? Maybe not, but I just noticed the other test projects had them
```
	  <RestorePackagesWithLockFile>true</RestorePackagesWithLockFile>
	  <RestoreLockedMode Condition="'$(ContinuousIntegrationBuild)' == 'true'">true</RestoreLockedMode>

File: shared/tests/Piipan.Shared.Cryptography.Tests/NacCryptographyClientTests.cs [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+            NacCryptographyClient nacCryptographyClient = new NacCryptographyClient(cryptographyClientMock.Object);
+
+            //Act
+            string decryptedValue = nacCryptographyClient.DecryptFromBase64String(stringToDecrypt);
+            
+            //Assert
+            Assert.Equal(stringToDecrypt, decryptedValue);
+        }
+
+        public class MockCryptographyClient : CryptographyClient
```
Comment (dborgen2) [inline shared/tests/Piipan.Shared.Cryptography.Tests/NacCryptographyClientTests.cs:64]: Nitpick: I think this Mock class isn't being used by anything and can be removed

Comment (bgfoshay) [reply shared/tests/Piipan.Shared.Cryptography.Tests/NacCryptographyClientTests.cs:64]: Yes, thank you. I missed removing this. I was using it originally and then found a better way to implement a client for integration testing.

File: shared/tests/Piipan.Shared.Cryptography.Tests/packages.lock.json (hunk 1 above)
Comment (bgfoshay) [reply shared/tests/Piipan.Shared.Cryptography.Tests/packages.lock.json:1302]: 👍

File: shared/tests/Piipan.Shared.Cryptography.Tests/Piipan.Shared.Cryptography.Tests.csproj (hunk 2 above)
Comment (bgfoshay) [reply shared/tests/Piipan.Shared.Cryptography.Tests/Piipan.Shared.Cryptography.Tests.csproj:4]: 👍

**Pull request merged**